### PR TITLE
Issue 61: fix for ubuntu and debian images

### PR DESCRIPTION
**Pull request by SilinPavel**
## Summary
This PR addresses BLOB service name resolution issue #61 on Ubuntu and Debian images

### Implemented steps
- Extend logging
- In launch.sh added section that force create symbol link from /bin/sh to /bin/bash

### CODE REVIEW

File: workflows/pipe-common/scripts/mount_storage.py [hunk 1]
```diff
@@ -277,6 +272,17 @@ def build_mount_command(self, params):
...
+    def __resolve_azure_blob_service_url(self):
+        # add resolved ip address for azure blob service to /etc/hosts (only once per account_name)
+        account_name, _, _ = self._get_credentials(self.storage)
+        command = 'etc_hosts_clear="$(sed -E \'/.*{account_name}.blob.core.windows.net.*/d\' /etc/hosts)" ' \
+                  '&& cat > /etc/hosts <<< "$etc_hosts_clear" ' \
+                  '&& getent hosts {account_name}.blob.core.windows.net >> /etc/hosts'.format(account_name=account_name)
+        exit_code, _, stderr = common.execute_cmd_command_and_get_stdout_stderr(command, silent=True)
+        if exit_code != 0:
+            Logger.warn(
+                'Something goes wrong with resolving azure blob service hosts to /etc/hosts: \n {}'.format(stderr))
```
Comment (sidoruka) [inline workflows/pipe-common/scripts/mount_storage.py]: 'Azure BLOB service hostname resolution and writing to /etc/hosts failed: ...'

File: workflows/pipe-common/scripts/mount_storage.py [hunk 2]
```diff
@@ -277,6 +272,17 @@ def build_mount_command(self, params):
...
                '{mount_options}'.format(**params)
 
+    def __resolve_azure_blob_service_url(self):
+        # add resolved ip address for azure blob service to /etc/hosts (only once per account_name)
+        account_name, _, _ = self._get_credentials(self.storage)
+        command = 'etc_hosts_clear="$(sed -E \'/.*{account_name}.blob.core.windows.net.*/d\' /etc/hosts)" ' \
+                  '&& cat > /etc/hosts <<< "$etc_hosts_clear" ' \
+                  '&& getent hosts {account_name}.blob.core.windows.net >> /etc/hosts'.format(account_name=account_name)
+        exit_code, _, stderr = common.execute_cmd_command_and_get_stdout_stderr(command, silent=True)
+        if exit_code != 0:
```
Comment (sidoruka) [inline workflows/pipe-common/scripts/mount_storage.py]: Is it the only place when shall handle shell errors?
E.g. we've experienced an issue with blobfuse installation for Debian (in the background it threw 404) and it was hard to troubleshoot without any log message

Comment (SilinPavel) [reply workflows/pipe-common/scripts/mount_storage.py]: In this script we already have checks for other places where shell is executed:
 - execute_mount method
 - create_directory method
 - execute_and_check_command method

Comment (sidoruka) [reply workflows/pipe-common/scripts/mount_storage.py]: In this case, what log will I get if blobfuse installation fails for whatever reason? E.g. when the following is executed:
https://github.com/epam/cloud-pipeline/blob/2dd2eae60f3fb8779a74e7751014fc64a79b79ae/workflows/pipe-common/scripts/mount_storage.py#L237

File: workflows/pipe-common/scripts/mount_storage.py [hunk 3]
```diff
@@ -281,7 +281,7 @@ def __resolve_azure_blob_service_url(self):
         exit_code, _, stderr = common.execute_cmd_command_and_get_stdout_stderr(command, silent=True)
         if exit_code != 0:
             Logger.warn(
-                'Something goes wrong with resolving azure blob service hosts to /etc/hosts: \n {}'.format(stderr))
+                'Azure BLOB service hostname resolution and writing to /etc/hosts failed:: \n {}'.format(stderr))
```
Comment (sidoruka) [inline workflows/pipe-common/scripts/mount_storage.py]: Would be nice if you double-check the spelling yourself... `::` -> `:`

File: workflows/pipe-common/scripts/mount_storage.py (hunk 2 above)
Comment (SilinPavel) [reply workflows/pipe-common/scripts/mount_storage.py]: lines from stdout and stderr will be printed by python process to the current shell, this won't be appear in a pipeline logs. Do you refer me to #151 and propose to resolve this issue in the current PR?

Comment (sidoruka) [reply workflows/pipe-common/scripts/mount_storage.py]: Yes

Comment (SilinPavel) [reply workflows/pipe-common/scripts/mount_storage.py]: ok, done, now stderr of all installation script (install_s3_fuse_goofys, install_azure_fuse_blobfuse etc) will be passed to pipeline as warn logs

**Pull request merged**